docs/reference/dep-008.md

```diff
@@ -0,0 +1,38 @@
++++
+dep = 8
+title = "Draft Tasks"
+authors = [ "Michelle Noorali <michelle.noorali@microsoft.com>" ]
+created = 2018-04-10
++++
+
+# Abstract
+
+This document describes draft `tasks`, how to define them, and when to use them.
+
+# Motivation
+
+Draft users may find themselves wanting to hook into different parts of the draft application lifecycle. For example, a user may want to set up or install a dependency in their cluster before running `draft up`, run commands in their application container after deploying it to their Kubernetes cluster, or perform cleanup steps after deleting their application. Draft tasks make it possible for users to define and run these types of actions.
+
+# Specification
+Tasks live in `.draft/tasks.toml` in an application's root directory. If a tasks.toml file is specified in a draft pack, running the `draft create` command will copy `tasks.toml` from the draft pack to `.draft/tasks.toml` in the application's root directory.
+
+The format of a tasks file is as follows:
+
+In `APP_ROOT/.draft/tasks.toml`:
+```toml
+[pre-up]
+hello = "echo hello world"
```

Comment:
Can we clarify what the name of the task represents? Is it just visual or does it serve a purpose?

Also, is there a deterministic order for tasks to be run? e.g. if I define 2 pre-up tasks "hello" and "goodbye", which would be run first?

Reply:
No that's perfect. Thank you! :)